homeassistant/components/recorder/purge.py

```diff
@@ -101,3 +59,67 @@ def purge_old_data(instance, purge_days: int, repack: bool) -> bool:
     except SQLAlchemyError as err:
         _LOGGER.warning("Error purging history: %s", err)
     return True
+
+
+def _select_event_ids_to_purge(session, purge_before):
```

Comment:
Could you add type annotations for the function definitions?

Reply:
done